### ISSUE TITLE
Typechecker does not check that "types" are actually types

### ISSUE THREAD

**Issue by sid-kap** [python/mypy#1231]
MyPy does not show an error for the following code:
```python
import typing

def foo(s: set[int]) -> int:
  return len(s)

def goo(l: list[int]) -> int:
  return len(l)
```

However, this code is not actually valid: it throws the following error at runtime:
```
Traceback (most recent call last):
  File "error.py", line 3, in <module>
    def foo(s: set[int]) -> int:
TypeError: 'type' object is not subscriptable
```

The first step to fixing this code is to change `set` to `Set` and `list` to `List`. This, however, leads to another runtime error:
```
Traceback (most recent call last):
  File "error.py", line 3, in <module>
    def foo(s: Set[int]) -> int:
NameError: name 'Set' is not defined
```

Finally, we need to add the following imports:
```python
from typing import Set, List
```
and the program finally runs without an error.

So there are two issues here:
1) mypy should catch when a "not subscriptable" thing (whatever that means) is written instead of an actual type
2) mypy should detect when types are used without being imported

**Comment by JukkaL** [comment]
Yeah, these are (known) bugs. I'm not sure if there is another issue created for this.

I believe that this only affects `List`, `Dict` and `Set` because they are special cased as aliases to `list`, `dict` and `set`, respectively. Mypy should not consider `List` and `list` interchangeable, as it currently does. However, `list` should still remain a valid type and be equivalent to `List[Any]`.

**Comment by gvanrossum** [comment]
This is probably a dupe of #999.
